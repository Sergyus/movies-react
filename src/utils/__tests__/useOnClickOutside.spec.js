import React from 'react';
import { create, act } from 'react-test-renderer';
import useOnClickOutside from '../useOnClickOutside';

describe('Body scroll lock work correctly', () => {
  const ref = {
    current: null,
  };
  const Component = () => {
    useOnClickOutside(ref);
    return <h1>Hello</h1>;
  };
  it('Body scroll lock not calls without ref', () => {
    let tree;
    act(() => {
      tree = create(<Component />);
    });
    const event = new MouseEvent('mousedown', {});
    act(() => {
      document.dispatchEvent(event);
    });
    expect(tree.toJSON()).toMatchSnapshot();
    tree.unmount();
  });
});
