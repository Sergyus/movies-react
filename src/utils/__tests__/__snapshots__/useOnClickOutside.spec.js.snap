// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Body scroll lock work correctly Body scroll lock not calls without ref 1`] = `
<h1>
  Hello
</h1>
`;
